dbcon/joblist/groupconcat.cpp

```diff
@@ -981,54 +1389,60 @@ const string GroupConcatOrderBy::toString() const
   return (baseStr + oss.str());
 }
 
-// GroupConcatNoOrder class implementation
-GroupConcatNoOrder::GroupConcatNoOrder()
- : fRowsPerRG(128), fErrorCode(ERR_AGGREGATION_TOO_BIG), fMemSize(0), fRm(NULL)
+uint64_t GroupConcatOrderBy::Hasher::operator()(const rowgroup::Row::Pointer& p) const
+{
+  Row& row = ts->row1;
+  row.setPointer(p);
+  return row.hash(colCount - 1);
```

Comment:
Rhetoric phrase.
This is what I like the most :)
Why do we hash using all columns `colCount - 1`. This assumes that RG contains only GB columns.

Reply:
The `colCount` value here is the exact number of columns that will be concatenated (and `distinct` operates on them)